jep/0000/README.adoc

```diff
@@ -0,0 +1,345 @@
+= JEP-0000: Opt-in Continuous Delivery of Jenkins Plugin Releases
+:toc: preamble
+:toclevels: 3
+ifdef::env-github[]
+:tip-caption: :bulb:
+:note-caption: :information_source:
+:important-caption: :heavy_exclamation_mark:
+:caution-caption: :fire:
+:warning-caption: :warning:
+endif::[]
+
+.Metadata
+[cols="1h,1"]
+|===
+| JEP
+| 0000
+
+| Title
+| Opt-in Continuous Delivery of Jenkins Plugin Releases from Trusted CI
+
+| Sponsor
+| link:https://github.com/daniel-beck[Daniel Beck]
+
+// Use the script `set-jep-status <jep-number> <status>` to update the status.
+| Status
+| Draft :speech_balloon:
+
+| Type
+| Process :bulb:
+
+| Created
+| Date (2019-05-15) :bulb:
+
+| BDFL-Delegate
+| TBD
+
+//
+//
+// Uncomment if there is an associated placeholder JIRA issue.
+//| JIRA
+//| :bulb: https://issues.jenkins-ci.org/browse/JENKINS-nnnnn[JENKINS-nnnnn] :bulb:
+//
+//
+// Uncomment if discussion will occur in forum other than jenkinsci-dev@ mailing list.
+//| Discussions-To
+//| :bulb: Link to where discussion and final status announcement will occur :bulb:
+//
+//
+// Uncomment if this JEP depends on one or more other JEPs.
+//| Requires
+//| :bulb: JEP-NUMBER, JEP-NUMBER... :bulb:
+//
+//
+// Uncomment and fill if this JEP is rendered obsolete by a later JEP
+//| Superseded-By
+//| :bulb: JEP-NUMBER :bulb:
+//
+//
+// Uncomment when this JEP status is set to Accepted, Rejected or Withdrawn.
+//| Resolution
+//| :bulb: Link to relevant post in the jenkinsci-dev@ mailing list archives :bulb:
+
+|===
+
+== Abstract
+
+This JEP introduces a new system, which enables Jenkins plugin maintainers to add Continuous 
+Delivery to their projects. It is:
+
+* Opt-in: plugin maintainers can choose to use this
+* Simple to configure: very little needs to be changed in order to enable CD
+* Trusted CI/CD: deployments are produced from a single source of truth, which is managed by 
+the Jenkins project
+
+== Specification
+
+=== Overview
+
+We'll use a short story to illustrate what Autorelease will look like when a plugin maintainer 
+named Karl decides that he wants to perform continuous delivery of his (fictional) Karl-Plugin.
+
+To do so, Karl will:
+
+1. Verify support. Karl's plugin is already in the 
+link:https://github.com/jenkinsci[jenkinsci Organization on GitHub], and is 
+using `buildPlugin` when built on ci.jenkins.io.
+
+2. Choose a branch from which to release. Karl decides he wants to release from the `master` branch, 
+as that is the simplest.
+
+3. Settle on a version numbering scheme. Becuase it's in fairly wide use across the Jenkins 
+ecosystem, Karl settles on link:https://semver.org/[semantic versioning].
+
+4. Add a file to the top directory of his repository called `Autorelease.yaml`. 
+
+5. A webhook is configured on Karl's repository on GitHub to point to `{the-webhook-receiver}`
```

Comment:
1. The webhook receiver is a computer, which does not exist yet. It'll act as a proxy of sorts, to take these incoming webhooks from commits, and send them to a secure CI server which lives on a VPN. This is in order to keep the CI server in a safe place.

2. My original thought was that the plugin maintainer would configure the webhook on a per-repository basis. That way, if Jane Java decides she wants to use CD on her `jane-plugin`, but release her `old-timey-plugin` from her laptop like she's always done, she can do so.

Maybe an organization-level webhook is the right approach. I hadn't thought of it.  Is there any risk of the webhook receiver being overrun with a huge influx of webhooks, every time someone commits to a repository in the jenkinsci organization? Not necessarily from the standpoint of overwhelming that receiver, but rather, from a logging standpoint. Let's say something goes wrong in this system. The infra team might have an easier time troubleshooting if the only webhooks getting sent are from individual repos, rather than from every repo in the org.

Reply:
> 1. The webhook receiver is…
> 2. My original thought was…

So write all this in the doc!

> Maybe an organization-level webhook is the right approach…

→ **Reasoning** section

Reply:
Done and done.